pkg/storage/formattedstore/batchwrite.go

```diff
@@ -176,35 +179,30 @@ func (p *pendingBatch) put(addBatch *batch) error {
 		return err
 	}
 
-	p.mutex.Lock()
-
 	p.values = append(p.values, models.VaultOperation{
 		Operation:         models.UpsertDocumentVaultOperation,
 		EncryptedDocument: *encryptedDocument,
 	})
 
-	p.mutex.Unlock()
-
 	return nil
 }
 
 func (p *pendingBatch) delete(addBatch *batch) error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
 	p.size++
 
 	id, err := p.formatter.GenerateEDVDocumentID(addBatch.keyID)
```

Comment:
Note: might be worth commenting on why p.formatter.GenerateEDVDocumentID depends on state.